wled00/FX.h

```diff
@@ -55,17 +53,17 @@
 /* each segment uses 52 bytes of SRAM memory, so if you're application fails because of
   insufficient memory, decreasing MAX_NUM_SEGMENTS may help */
 #ifdef ESP8266
-  #define MAX_NUM_SEGMENTS    12
+  #define MAX_NUM_SEGMENTS    18
   /* How many color transitions can run at once */
   #define MAX_NUM_TRANSITIONS  8
   /* How much data bytes all segments combined may allocate */
-  #define MAX_SEGMENT_DATA  2048
+  #define MAX_SEGMENT_DATA  4096
```

Comment:
Pretty sure this will run out of memory with 1600 LEDs defined. Need to test though, but I'm not sure I'm comfortable with allocating this much. Falling back to solid is better than becoming unresponsive due to out of memory.

Reply:
😄 
Are you sure you want to run 1600 LEDs on 8266?
I tested it with 1200 LEDs defined and 13 segments (with a few usermods). Beyond 13th segment ESP would run out of memory. But if you keep LED count to a reasonable 600 then it is still fine.

Reply:
I never would 😄 way too little FPS and yeah, you run out of memory really quickly.
But yeah, I have "allowed" up to 5k for the LED buffer, which is around 1600.
Maybe making the effect data memory limit dynamic would be a good compromise here. 6-8k for LED buffer and data combined. With "only" 600 LEDs, you'll have a massive 4.2-6.2k for data, which shrinks to 1-3k if you "need to" max out the LED count. Best of both worlds :)

Reply:
Perhaps a redefinition of what someone should expect from 8266 would be in order.
We made many changes that indeed push 8266 to the limits (if not beyond) so reducing the maximum number of LEDs would not be too bad (1200 perhaps, 2 strips of 600 each?).
I may have exaggerated with maximum segment numbers since a few users requested it, but none of them used maximum number of LEDs.

Perhaps we could make a maximum number of segments dependant on maximum number of LEDs?

Reply:
Can these parameters not be something left to platformio.ini to define, based on the device type? some code to do a sanity check for "safe" limits would still be required, of course.  I would expect anyone wanting to push the limits in terms of LEDs and segments shouldn't be balking at the extra cost of ESP32, but I sense there are still quite a few using this to mod devices with very lightweight devices built in (I do both)